src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+    }
+
+    if (func_801378B8(&this->skelAnime, 28.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_EAT);
+        if (this->actor.child != NULL) {
+            Actor_MarkForDeath(this->actor.child);
+            this->actor.child = NULL;
+        }
+        this->actor.flags &= ~0x10;
+    } else if (this->skelAnime.animPlaybackSpeed > -0.5f) {
+        this->unk_302--;
+        if (this->unk_302 == 10) {
+            for (i = 10; i >= 0; i--) {
+                sp84.x = randPlusMinusPoint5Scaled(10.0f);
+                sp84.y = randPlusMinusPoint5Scaled(10.0f);
+                sp84.z = randPlusMinusPoint5Scaled(10.0f);
+                sp78.x = sp84.x * -0.1f;
+                sp78.y = sp84.y * -0.1f;
+                sp78.z = sp84.z * -0.1f;
+                sp64.x = this->collider1Elements[0].dim.worldSphere.center.x + sp84.x;
+                sp64.y = this->collider1Elements[0].dim.worldSphere.center.y + sp84.y;
+                sp64.z = this->collider1Elements[0].dim.worldSphere.center.z + sp84.z;
+                func_800B0EB0(globalCtx, &sp64, &sp84, &sp78, &this->unk_32C, &this->unk_330, this->unk_334 * 400.0f,
+                              this->unk_334 * 10.0f, 10);
+            }
+
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+            if (this->actor.colChkInfo.health < 5) {
+                this->actor.colChkInfo.health = 0;
+                func_800BBA88(globalCtx, &this->actor);
+            } else {
+                this->actor.colChkInfo.health -= 4;
+            }
+            func_800BCB70(&this->actor, 0x4000, 0x78, 0, 8);
+        }
+    } else if (func_801378B8(&this->skelAnime, 24.0f)) {
+        this->unk_302--;
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+    }
+
+    if (this->skelAnime.animCurrentFrame <= 27.0f) {
+        if ((this->skelAnime.animCurrentFrame <= 25.0f) && (this->unk_302 < 11)) {
+            func_80876DC4(this, globalCtx);
+        } else {
+            sp5E = this->unk_334 * 50.0f;
+            sp5C = this->unk_334 * 5.0f;
+            Math_Vec3f_Copy(&sp64, &this->unk_348[0]);
+            func_800B0DE0(globalCtx, &sp64, &D_801D15B0, &D_80879360, &D_8087936C, &D_8087936C, sp5E, sp5C);
+            sp64.x -= Math_CosS(this->actor.shape.rot.y) * 6.0f * this->unk_334;
+            sp64.z += Math_SinS(this->actor.shape.rot.y) * 6.0f * this->unk_334;
+            func_800B0DE0(globalCtx, &sp64, &D_801D15B0, &D_80879360, &D_8087936C, &D_8087936C, sp5E, sp5C);
+            sp64.x = (2.0f * this->unk_348[0].x) - sp64.x;
+            sp64.z = (2.0f * this->unk_348[0].z) - sp64.z;
+            func_800B0DE0(globalCtx, &sp64, &D_801D15B0, &D_80879360, &D_8087936C, &D_8087936C, sp5E, sp5C);
+        }
+    }
+
+    this->unk_314.z = Math_SinS((this->actor.colorFilterTimer << 0xC) >> 1) + 1.0f;
+    this->unk_314.x = Math_SinS(this->actor.colorFilterTimer << 0xC) + 1.0f;
+    this->unk_314.y = this->unk_314.z;
```

Comment:
This looks pretty weird lol. Could it be a temp with `* 0x1000`, and then `/2` in the first sine?

Reply:
Yeah it's weird, but I couldn't fix it any other way. Temp has regalloc differences. A single temp used twice loses the second colorFilterTimer load. A single temp assigned twice uses s0 which is wrong, 2 temps make the stack too big and still have regalloc differences.

I can use 0x1000 for the multiply, but the >> 1 can't be replaced.